src/openms/include/OpenMS/ANALYSIS/MAPMATCHING/QTClusterFinder.h

```diff
@@ -147,12 +162,41 @@ namespace OpenMS
     void setParameters_(double max_intensity, double max_mz);
 
     /// Generates a consensus feature from the best cluster and updates the clustering
-    void makeConsensusFeature_(std::list<QTCluster>& clustering,
+    bool makeConsensusFeature_(Heap& cluster_heads,
                                ConsensusFeature& feature,
-                               ElementMapping& element_mapping, Grid&);
+                               ElementMapping& element_mapping,
+                               Grid& grid,
+                               const std::vector<Handle>& handles);
 
     /// Computes an initial QT clustering of the points in the hash grid
-    void computeClustering_(Grid& grid, std::list<QTCluster>& clustering);
+    void computeClustering_(Grid& grid,
+                            Heap& cluster_heads,
+                            std::vector<QTCluster::Data_>& cluster_data,
+                            std::vector<Handle>& handles);
+
+    /// Removes ids of current top cluster in the heap from element mapping and pops the heap
+    void removeTopFromHeap_(Heap& cluster_heads,
+                            const QTCluster& cluster,
+                            ElementMapping& element_mapping);
+    
+    /// creates a consensus feature from the given elements
+    void createConsensusFeature_(ConsensusFeature& feature, double quality, 
+                                 const ClusterElementsMap& elements);
+
+    /** 
+     * @brief update the clustering:
+     *
+     * 1. remove current "best" cluster from list
+     * 2. update all clusters accordingly by removing already used elements
+     * 3. Invalidate elements whose central has been used already
```

Comment:
```
1. remove current best cluster from the heap
2. update all clusters accordingly by removing neighbors used by the current best
3. invalidate clusters whose center has been used by the current best
```

I think "clusters" instead of "elements" is good, but "have been used by the current best" is a better hint towards what's really happening than " have been removed".